vsomeip-sys/README.md

```diff
@@ -0,0 +1,57 @@
+# vsomeip-sys
+
+## What is it?
+
+A fairly basic wrapper around the essentials needed to make a uProtocol uTransport implementation for SOME/IP based on top of the C++ vsomeip library.
+
+## How do I build it?
+
+1. Ensure you have a Rust toolchain installed
+2. Ensure you have the vsomeip library installed
+
+Then,
+
+```bash
+VSOMEIP_LIB_PATH=<path/to/vsomeip/lib> GENERIC_CPP_STDLIB_PATH=<path/to/generic/cpp/stdlib> ARCH_SPECIFIC_CPP_STDLIB_PATH=<path/to/arch_specific/cpp/stdlib> cargo build
+```
+
+If you are running a Linux-based OS running x86_64, it's possible these look like:
+
+```
+GENERIC_CPP_STDLIB_PATH = /usr/include/c++/11
+ARCH_SPECIFIC_CPP_STDLIB_PATH=/usr/include/x86_64-linux-gnu/c++/11
+```
+
+## Running a binary built with vsomeip-sys
+
+You will need to ensure that `LD_LIBRARY_PATH` includes the path to your vsomeip library install by:
+
+```bash
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:<path/to/vsomeip/lib> ./your_binary_built_with_vsomeip-sys
+```
+
+or by permanently modifying your `LD_LIBRARY_PATH` to include the path to your vsomeip library install.
+
+This is required because of symlinking that's performed by the vsomeip library install (which is typical).
+
+## Supplying your own vsomeip includes
+
+You may modify the build command above to include setting the environment variable:
+
+```
+VSOMEIP_INCLUDE_PATH=<path/to/vsomeip/include>
+```
+
+`VSOMEIP_INCLUDE_PATH` must point to the folder which has the `vsomeip` folder within it containing the vsomeip includes.
```

Comment:
My understanding is that we _always_ require the user to have the vsomeip shared library installed, i.e. we do not create it as part of the vsomeip-sys build.
If that is the case, then I would suggest to also require the user to _always_ provide the (matching) header files. Otherwise we might run into situations where the user provides the lib in version A and we download the source code in version B. This doesn't seem to pose any additional effort for the user because he needs to build vsomeip from source anyway, as there is no binary distribution and as part of the build/install process the vsomeip headers will also be installed, right?

Reply:
Added the ability to do this in the latest commit. Looking for feedback